### PR TITLE
Nancy's Homework 

### PR DESCRIPTION
**Pull request by nyhyang**

### CODE REVIEW

File: D04_ex00.py [hunk 1]
```diff
@@ -11,10 +11,27 @@
...
 # Body
 
+random_integer = random.randint(1, 25)
+
+
+guesses_left = 5
+while guesses_left > 0:
+	try:
+		number = float(input('Enter a number: '))
+		if int(number) == random_integer:
```
Comment (danielsgriffin) [inline D04_ex00.py]: Is this right? If random_integer is 1 and my guess is 1.5 should that be correct?

File: HW04_ch07_ex02.py [hunk 2]
```diff
@@ -21,15 +21,28 @@
 
 ###############################################################################
 # Imports
-
+import math
 
 # Body
+def eval_loop():
```
Comment (danielsgriffin) [inline HW04_ch07_ex02.py]: "return the value of the last expression it evaluated"

File: HW04_ch08_ex00.py [hunk 3]
```diff
@@ -17,17 +17,31 @@
 
 ###############################################################################
 # Imports
-
+import math
```
Comment (danielsgriffin) [inline HW04_ch08_ex00.py]: why?

File: HW04_ch08_ex04.py [hunk 4]
```diff
@@ -16,6 +16,8 @@
 def any_lowercase1(s):
     """Explain what is wrong, if anything, here.
     """
+    #.islower will only return tru if ALL the character is in lower case but if
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: not quite.

File: HW04_ch08_ex04.py [hunk 5]
```diff
@@ -26,6 +28,8 @@ def any_lowercase1(s):
 def any_lowercase2(s):
     """Explain what is wrong, if anything, here.
     """
+    #'c' is not equal to c, it's not refering to check the items in s, it's checkecking if 
+    # if there is actual c inside s. It will return since c is in lowercase.
```
Comment (danielsgriffin) [inline HW04_ch08_ex04.py]: "if there is actual c inside s"
false

File: HW04_ch08_ex05.py [hunk 6]
```diff
@@ -1,4 +1,46 @@
-# Structure this script entirely on your own.
-# See Chapter 8: Strings Exercise 5 for the task.
-# Please do provide function calls that test/demonstrate your
-# function.
+# # Structure this script entirely on your own.
+# # See Chapter 8: Strings Exercise 5 for the task.
+# # Please do provide function calls that test/demonstrate your
```
Comment (danielsgriffin) [inline HW04_ch08_ex05.py]: (1) Note this: `rotated += chr(letter)` (adding to your rotated initialization of ' '
Caused this issue:
```
Book Examples:
  X  'A'     by 3   => got: ' D'    expected: 'D'
  X  'Z'     by 1   => got: ' ['    expected: 'A'
  X  'cheer' by 7   => got: ' jolly' expected: 'jolly'
  X  'melon' by -10 => got: ' c[bed' expected: 'cubed'
  X  'IBM'   by -1  => got: ' HAL'  expected: 'HAL'

Daniel's Tests:
  X  'abc'   by 1   => got: ' bcd'  expected: 'bcd'
  X  'xyz'   by 1   => got: ' yz{'  expected: 'yza'
  X  'ABC'   by 1   => got: ' BCD'  expected: 'BCD'
  X  'XYZ'   by 1   => got: ' YZ['  expected: 'YZA'
  X  'Az'    by 1   => got: ' B{'   expected: 'Ba'
  X  'Aa'    by 26  => got: ' [{'   expected: 'Aa'
  X  'abc'   by -1  => got: ' `ab'  expected: 'zab'
  X  'xyz'   by -1  => got: ' wxy'  expected: 'wxy'
  X  'ABC'   by -1  => got: ' @AB'  expected: 'ZAB'
  X  'XYZ'   by -1  => got: ' WXY'  expected: 'WXY'
  X  'Az'    by -1  => got: ' @y'   expected: 'Zy'
  X  'Aa'    by -26 => got: " 'G"   expected: 'Aa'
```

Which isn't nearly as bad if that were fixed:
```
Book Examples:
 OK  'A'     by 3   => got: 'D'     expected: 'D'
  X  'Z'     by 1   => got: '['     expected: 'A'
 OK  'cheer' by 7   => got: 'jolly' expected: 'jolly'
  X  'melon' by -10 => got: 'c[bed' expected: 'cubed'
 OK  'IBM'   by -1  => got: 'HAL'   expected: 'HAL'

Daniel's Tests:
 OK  'abc'   by 1   => got: 'bcd'   expected: 'bcd'
  X  'xyz'   by 1   => got: 'yz{'   expected: 'yza'
 OK  'ABC'   by 1   => got: 'BCD'   expected: 'BCD'
  X  'XYZ'   by 1   => got: 'YZ['   expected: 'YZA'
  X  'Az'    by 1   => got: 'B{'    expected: 'Ba'
  X  'Aa'    by 26  => got: '[{'    expected: 'Aa'
  X  'abc'   by -1  => got: '`ab'   expected: 'zab'
 OK  'xyz'   by -1  => got: 'wxy'   expected: 'wxy'
  X  'ABC'   by -1  => got: '@AB'   expected: 'ZAB'
 OK  'XYZ'   by -1  => got: 'WXY'   expected: 'WXY'
  X  'Az'    by -1  => got: '@y'    expected: 'Zy'
  X  'Aa'    by -26 => got: "'G"    expected: 'Aa'
```

Do you know how to fix it?

**Comment by danielsgriffin** [comment]
Good job. Make sure you test your functions.

File: HW04_string2.py [hunk 7]
```diff
@@ -29,11 +34,17 @@ def verbing(s):
...
+
+
 def not_bad(s):
     # +++your code here+++
-    return
+    snot_index = s.find('not')
+    sbad_index= s.find('bad')
 
+    if sbad_index > snot_index:
+        s = s.replace(s[snot_index:(sbad_index+3)], 'good')
```
Comment (ankeetshankar) [inline HW04_string2.py]: Nice Small code !!

File: HW04_string2.py [hunk 8]
```diff
@@ -43,7 +54,26 @@ def not_bad(s):
 #  a-front + b-front + a-back + b-back
 def front_back(a, b):
     # +++your code here+++
-    return
+    alength = len(a)
```
Comment (ankeetshankar) [inline HW04_string2.py]: Code works according to test case.

**Pull request closed without merging**